src/ui/common/form/RoleSelectRenderer.js

```diff
@@ -0,0 +1,126 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { InputGroup, Button, Col } from 'patternfly-react';
+import {
+  FormattedMessage,
+  injectIntl,
+  intlShape,
+  defineMessages,
+} from 'react-intl';
+
+class RoleSelectRenderer extends Component {
+  constructor(props) {
+    super(props);
+    this.pushField = this.pushField.bind(this);
+    this.renderTags = this.renderTags.bind(this);
+    this.select = null;
+  }
+
+  pushField() {
+    if (!this.select || !this.select.value) {
+      return;
+    }
+    const {
+      selectedValues, fields,
+    } = this.props;
+
+    if (this.select.value && !selectedValues.includes(this.select.value)) {
+      fields.push(this.select.value);
+    }
+  }
+
+  renderTags() {
+    const {
+      selectedValues, fields, labelKey, valueKey, options,
+    } = this.props;
+    return selectedValues.map((value, i) => (
+      <div key={value}>
+        <h3><FormattedMessage id="cms.contenttype.labelrole.assigned" /></h3>
+        <hr />
+        <Col xs={4}>
+          <p>
+            {options.find(opt => opt[valueKey] === value)[labelKey]}
+          </p>
+        </Col>
+        <Col xs={8}>
+          <Button
+            bsStyle="danger"
+            className="btn btn-danger RoleSelectRenderer__remove--btn"
+            onClick={() => fields.remove(i)}
+          >
+            <FormattedMessage id="cms.label.delete" />
+          </Button>
+        </Col>
+      </div>
+    ));
+  }
```

Comment:
I had almost same like this and Nic told me that "no rendering methods should exist in the class other than default render method". He also advised me to either move this code into render and assign it to a variable or create a separate mini component and import it into render